recipes/lora_finetune_single_device.py

```diff
@@ -361,6 +369,17 @@ def save_checkpoint(self, epoch: int) -> None:
             intermediate_checkpoint=(epoch + 1 < self.total_epochs),
         )
 
+    def _should_update_weights(self, current_iteration: int) -> bool:
+        """
+        Determines whether the weights should be updated on the current iteration or not.
+        True is returned either if we've accumulated gradients for enough steps or if this
+        is the last step in the epoch.
```

Comment:
nit: I don't think the second half applies here anymore. (I know it's the same in other recipes too)
```suggestion
        True is returned either if we've accumulated gradients for enough steps.
```

Reply:
Thanks for catching!